R/infill_crits_interface.R

```diff
@@ -0,0 +1,366 @@
+#' @rdname infillcrits
+makeMBOInfillCriterionStandardError = function() {
+  makeMBOInfillCriterion(
+    fun = function(points, models, control, par.set, design, iter, attributes = FALSE) {
+      -predict(models[[1L]], newdata = points)$data$se
+    },
+    name = "Standard error",
+    id = "sd"
+  )
+}
+
+#' @export
+#' @rdname infillcrits
+makeMBOInfillCriterionEI = function() {
+  makeMBOInfillCriterion(
+    fun = function(points, models, control, par.set, design, iter, attributes = FALSE) {
+      model = models[[1L]]
+      maximize.mult = ifelse(control$minimize, 1, -1)
+      y = maximize.mult * design[, control$y.name]
+      p = predict(model, newdata = points)$data
+      p.mu = maximize.mult * p$response
+      p.se = p$se
+      y.min = min(y)
+      d = y.min - p.mu
+      xcr = d / p.se
+      xcr.prob = pnorm(xcr)
+      xcr.dens = dnorm(xcr)
+      ei = d * xcr.prob + p.se * xcr.dens
+      # FIXME: magic number
+      # if se too low set 0 (numerical problems), negate due to minimization
+      res = ifelse(p.se < 1e-6, 0, -ei)
+      if (attributes) {
+        res = setAttribute(res, "crit.components", data.frame(se = p$se, mean = p$response))
+      }
+      return(res)
+    },
+    name = "Expected improvement",
+    id = "ei",
+    components = c("se", "mean"),
+    minimize = FALSE
+  )
+}
+
+#' @export
+#' @rdname infillcrits
+makeMBOInfillCriterionCB = function(cb.lambda = 1, cb.inflate.se = FALSE) {
+  assertNumber(cb.lambda, lower = 0)
+  assertFlag(cb.inflate.se)
+  makeMBOInfillCriterion(
+    fun = function(points, models, control, par.set, design, iter, attributes = FALSE) {
+      force(cb.lambda)
+      force(cb.inflate.se)
+      model = models[[1L]]
+      maximize.mult = ifelse(control$minimize, 1, -1)
+      p = predict(model, newdata = points)$data
+      if (cb.inflate.se) {
+        r.response = diff(range(p$response))
+        r.se = diff(range(p$se))
+        tol = .Machine$double.eps^0.5
+        if (r.response < tol)
+          return(r.se)
+        if (r.se < tol)
+          return(r.response)
+        inflate = r.response / r.se
+      } else {
+        inflate = 1
+      }
+      res = maximize.mult * p$response - inflate * cb.lambda * p$se
+      if (attributes) {
+        res = setAttribute(res, "crit.components",
+          data.frame(se = p$se, mean = p$response, lambda = cb.lambda))
+      }
+      return(res)
+    },
+    name = "Lower confidence bound",
+    id = "cb",
+    components = c("se", "mean", "lambda")
+  )
+}
+
+#' @export
+#' @rdname infillcrits
+makeMBOInfillCriterionAEI = function(aei.use.nugget = FALSE) {
+  assertFlag(aei.use.nugget)
+  force(aei.use.nugget)
+
+  makeMBOInfillCriterion(
+    fun = function(points, models, control, par.set, design, iter, attributes = FALSE) {
+      model = models[[1L]]
+      maximize.mult = ifelse(control$minimize, 1, -1)
+      p = predict(model, newdata = points)$data
+      p.mu = maximize.mult * p$response
+      p.se = p$se
```

Comment:
The above 5 lines are used in most infill criterias. Perhaps add a small helper function?